src/Ocelot/Logging/AspDotNetLogger.cs

```diff
@@ -1,93 +1,66 @@
-            var previousRequestId = GetOcelotPreviousRequestId();
-
-            var state = $"requestId: {requestId}, previousRequestId: {previousRequestId}, message: {message}";
-
-            _logger.Log(LogLevel.Critical, default, state, exception, _func);
-        }
-
-        private string GetOcelotRequestId()
-        {
-            var requestId = _scopedDataRepository.Get<string>("RequestId");
-
-            return requestId == null || requestId.IsError ? "no request id" : requestId.Data;
-        }
-
-        private string GetOcelotPreviousRequestId()
-        {
-            var requestId = _scopedDataRepository.Get<string>("PreviousRequestId");
-
-            return requestId == null || requestId.IsError ? "no previous request id" : requestId.Data;
-        }
-    }
+
+namespace Ocelot.Logging;
+
+public class AspDotNetLogger : IOcelotLogger
+{
+    private readonly ILogger _logger;
+    private readonly IRequestScopedDataRepository _scopedDataRepository;
+    private readonly Func<string, Exception, string> _func;
+
+    public AspDotNetLogger(ILogger logger, IRequestScopedDataRepository scopedDataRepository)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _scopedDataRepository = scopedDataRepository;
+        _func = (state, exception) => exception == null ? state : $"{state}, exception: {exception}";
+    }
+
+    public void LogTrace(string message) => LogTrace(() => message);
+    public void LogTrace(Func<string> messageFactory) => WriteLog(LogLevel.Trace, messageFactory);
+
+    public void LogDebug(string message) => LogDebug(() => message);
+    public void LogDebug(Func<string> messageFactory) => WriteLog(LogLevel.Debug, messageFactory);
+
+    public void LogInformation(string message) => LogInformation(() => message);
+    public void LogInformation(Func<string> messageFactory) => WriteLog(LogLevel.Information, messageFactory);
+
+    public void LogWarning(string message) => LogWarning(() => message);
+    public void LogWarning(Func<string> messageFactory) => WriteLog(LogLevel.Warning, messageFactory);
+
+    public void LogError(string message, Exception exception) => LogError(() => message, exception);
+    public void LogError(Func<string> messageFactory, Exception exception) => WriteLog(LogLevel.Error, messageFactory, exception);
+
+    public void LogCritical(string message, Exception exception) => LogCritical(() => message, exception);
+    public void LogCritical(Func<string> messageFactory, Exception exception) => WriteLog(LogLevel.Critical, messageFactory, exception);
+
+    private string GetOcelotRequestId()
+    {
+        var requestId = _scopedDataRepository.Get<string>("RequestId");
+
+        return requestId == null || requestId.IsError ? "no request id" : requestId.Data;
+    }
+
+    private string GetOcelotPreviousRequestId()
+    {
+        var requestId = _scopedDataRepository.Get<string>("PreviousRequestId");
```

Comment:
I haven't checked that, it's part of the current implementation.

Reply:
Singleton, aha ok, using http context items to pass parameters.
```c#
public Response<T> Get<T>(string key)
        {
            if (_httpContextAccessor.HttpContext == null || _httpContextAccessor.HttpContext.Items == null)
            {
                return new ErrorResponse<T>(new CannotFindDataError($"Unable to find data for key: {key} because HttpContext or HttpContext.Items is null"));
            }

            if (_httpContextAccessor.HttpContext.Items.TryGetValue(key, out var obj))
            {
                var data = (T)obj;
                return new OkResponse<T>(data);
            }

            return new ErrorResponse<T>(new CannotFindDataError($"Unable to find data for key: {key}"));
        }
```